### PR TITLE
[clariden] pytorch ddp CE test

### PR DESCRIPTION
**Pull request by henrique**

### CODE REVIEW

**Comment by teojgo** [comment]
cscs-ci run alps-cpe-tests

**Comment by henrique** [comment]
cscs-ci run alps-cpe-tests

**Comment by jgphpc** [comment]
cscs-ci run alps-cpe-tests

File: checks/apps/pytorch/pytorch_amd.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import os
+import sys
+import pathlib
+import reframe as rfm
+import reframe.utility.sanity as sn
+from pytorch_test_base import PyTorchTestBase
+
+sys.path.append(str(pathlib.Path(__file__).parent.parent.parent / 'mixins'))
+from container_engine import ContainerEngineMixin  # noqa: E402
+
```
Comment (teojgo) [inline checks/apps/pytorch/pytorch_amd.py:10]: Remove one blank line

File: checks/apps/pytorch/pytorch_nvidia.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import os
+import sys
+import pathlib
+import reframe as rfm
+import reframe.utility.sanity as sn
+from pytorch_test_base import PyTorchTestBase
+
+sys.path.append(str(pathlib.Path(__file__).parent.parent.parent / 'mixins'))
+from container_engine import ContainerEngineMixin  # noqa: E402
+
```
Comment (teojgo) [inline checks/apps/pytorch/pytorch_nvidia.py:10]: Remove one blank line

File: checks/apps/pytorch/pytorch_nvidia.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    valid_systems = ['+nvgpu +sarus']
+    platform = 'Sarus'
+
+    @run_before('run')
+    def set_container_variables(self):
+        self.container_platform = self.platform
+        self.container_platform.command = self.executable
+        self.container_platform.image = 'nvcr.io/nvidia/pytorch:22.12-py3'  # cuda11.8 pt1.14.0
+        self.job.launcher.options.append('--mpi=pmi2')
+
```
Comment (teojgo) [inline checks/apps/pytorch/pytorch_nvidia.py:24]: Remove one blank line

File: checks/apps/pytorch/pytorch_nvidia.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import os
```
Comment (teojgo) [inline checks/apps/pytorch/pytorch_nvidia.py:1]: This is not used

**Review by teojgo — CHANGES_REQUESTED** [review]

**Comment by henrique** [comment]
cscs-ci run alps-cpe-tests

**Comment by henrique** [comment]
the AMD tests were failing repeatedly and randomly, so I removed them for now (I will create a separated PR for that)
all remaining [NVIDIA] tests passed
cheers!

**Comment by henrique** [comment]
ok, finally everything passed!
the amd tests have to go on another PR, as they seemed a bit unstable.
Cheers!

File: checks/apps/pytorch/pytorch_nvidia.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            self.container_env_table = {
+                'annotations.com.hooks': {
+                    'aws_ofi_nccl.enabled': 'true',
+                    'aws_ofi_nccl.variant': cuda_major,
+                },
+            }
+
+@rfm.simple_test
+class PyTorchDdpCeNvlarge(PyTorchDdpCeNv):
+    aws_ofi_nccl = parameter([True, False])
```
[the remark this answers is not in the document]
Comment (henrique) [reply checks/apps/pytorch/pytorch_nvidia.py:52]: I thought it didn't hurt and it might be useful data in case of a regression

**Pull request merged**